common/membership/hashring.go

```diff
@@ -239,92 +232,86 @@ func (r *ringpopHashring) refreshNoLock() error {
 
 	ring := emptyHashring()
 	for _, addr := range addrs {
-		host := NewHostInfo(addr, r.getLabelsMap())
+		host := NewHostInfo(addr)
 		ring.AddMembers(host)
 	}
 
-	r.membersMap = newMembersMap
-	r.lastRefreshTime = time.Now()
-	r.ringValue.Store(ring)
-	r.logger.Info("Current reachable members", tag.Addresses(addrs))
+	r.members.keys = newMembersMap
+	r.members.refreshed = time.Now()
+	r.value.Store(ring)
+	r.logger.Info(
+		fmt.Sprintf("refreshed ring members for %s", r.service),
+		tag.Addresses(addrs),
+	)
 	return nil
 }
 
-func (r *ringpopHashring) emitEvent(
-	rpEvent events.RingChangedEvent,
-) {
-
-	// Marshall the event object into the required type
-	event := &ChangedEvent{}
-	for _, addr := range rpEvent.ServersAdded {
-		event.HostsAdded = append(event.HostsAdded, NewHostInfo(addr, r.getLabelsMap()))
-	}
-	for _, addr := range rpEvent.ServersRemoved {
-		event.HostsRemoved = append(event.HostsRemoved, NewHostInfo(addr, r.getLabelsMap()))
-	}
-	for _, addr := range rpEvent.ServersUpdated {
-		event.HostsUpdated = append(event.HostsUpdated, NewHostInfo(addr, r.getLabelsMap()))
-	}
-
-	// Notify subscribers
-	r.smu.RLock()
-	defer r.smu.RUnlock()
-
-	for name, ch := range r.subscribers {
```

Comment:
@mantas-sidlauskas I was looking at why we never see [Ring membership changed](https://github.com/uber/cadence/blob/bae97d5da98026cabb7af95ea90c96da68e16449/service/history/shard/controller.go#L372) log and looks like after this refactoring we stopped notifying the subscribers. Was this intentional or missed?